app/services/migration/induction_sequence_exporter.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Migration
+  # Exports induction sequence analysis results to CSV format
+  class InductionSequenceExporter
+    DEFAULT_FILE_NAME = 'induction_sequences_analysis'
+
+    def initialize(results)
+      @results = results
+    end
+
+    def export_to_csv(filename: nil)
+      file_path = prepare_file_path(filename)
+
+      Rails.logger.info("InductionSequenceExporter: Exporting analysis to CSV: #{file_path}")
+
+      CSV.open(file_path, "w", write_headers: true, headers: csv_headers) do |csv|
+        @results.each do |result|
+          write_result_to_csv(csv, result)
+        end
+      end
+
+      Rails.logger.info("InductionSequenceExporter: Export completed: #{file_path}")
+      file_path.to_s
+    end
+
+    def csv_headers
```

Comment:
Because there are lots of columns here it might be easier (if a little less efficient) to create an array of hashes and pass them in one at a time, something like this:

```ruby
data = [{a: "b", c: "d", e: "f"}, {a: "b", c: "d", e: "f"}, {a: "b", c: "d", e: "f"}]

puts CSV.generate(headers: data.first.keys, write_headers: true) { |csv| data.each { |r| csv << r } }
```

Otherwise it's really tricky to tell if they line up correctly!